test/helpers/kubectl.go

```diff
@@ -1842,6 +1880,78 @@ func (kub *Kubectl) CiliumServicePreFlightCheck() error {
 	return nil
 }
 
+func (kub *Kubectl) DeployETCDOperator() error {
+	const etcdOperatorPath = "../examples/kubernetes/addons/etcd-operator/"
+	deployFile := func(filename string) error {
+		cmdRes := kub.Apply(GetFilePath(etcdOperatorPath + filename))
+		if !cmdRes.WasSuccessful() {
+			return fmt.Errorf("Unable to deploy descriptor of etcd-operator %s: %s", filename, cmdRes.CombineOutput().String())
+		}
+		return nil
+	}
+
+	kub.Exec(fmt.Sprintf("%s '%s'", GetFilePath(etcdOperatorPath+"tls/certs/gen-cert.sh"), "cluster.local"))
+	kub.Exec(GetFilePath(etcdOperatorPath + "tls/deploy-certs.sh"))
+
+	err := deployFile("00-crd-etcd.yaml")
+	if err != nil {
+		return err
+	}
+	err = deployFile("cilium-etcd-cluster.yaml")
+	if err != nil {
+		return err
+	}
+	err = deployFile("cilium-etcd-sa.yaml")
+	if err != nil {
+		return err
+	}
+	err = deployFile("cluster-role-binding-template.yaml")
+	if err != nil {
+		return err
+	}
+	err = deployFile("cluster-role-template.yaml")
+	if err != nil {
+		return err
+	}
+	return deployFile("deployment.yaml")
+}
+
+func (kub *Kubectl) DeleteETCDOperator() error {
+	const etcdOperatorPath = "../examples/kubernetes/addons/etcd-operator/"
+	deleteFile := func(filename string) error {
+		cmdRes := kub.Delete(GetFilePath(etcdOperatorPath + filename))
+		if !cmdRes.WasSuccessful() {
+			return fmt.Errorf("Unable to deploy descriptor of etcd-operator %s: %s", filename, cmdRes.CombineOutput().String())
+		}
+		return nil
+	}
+
+	kub.Exec(fmt.Sprintf("%s '%s'", GetFilePath(etcdOperatorPath+"tls/certs/gen-cert.sh"), "cluster.local"))
+	kub.Exec(GetFilePath(etcdOperatorPath + "tls/deploy-certs.sh"))
+
+	err := deleteFile("deployment.yaml")
+	if err != nil {
+		return err
+	}
+	err = deleteFile("cluster-role-template.yaml")
```

Comment:
On delete, all should be deleted, does not matter if fails or not, and return the error at the end, if not the leftovers config will be in there.